File: TripPlannerAPI/TripPlannerDAL/Entity/Gebruiker.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BCrypt.Net;

namespace TripPlannerDAL.Entity
{
    public class Gebruiker
    {
        [Key]
        public int userId { get; set; }
        public string voornaam { get; set; }
        public string achternaam { get; set; }
        public string mail { get; set; }
        public string hashedwachtwoord { get; set; }
        public bool isAdmin { get; set; }
        public bool isActief { get; set; }
        public string adres { get; set; }
        public int? plaatsId { get; set; }
        public Plaats? plaats { get; set; }
        public ICollection<UserTrip>? userTrips { get; set; }

        public void HashPassword(string password)
        {
            // Hash the password and store it in the hashedwachtwoord property
            hashedwachtwoord = BCrypt.Net.BCrypt.HashPassword(password);
        }

        public bool VerifyPassword(string password)
        {
            // Compare the provided password with the hashed password
            return BCrypt.Net.BCrypt.Verify(password, hashedwachtwoord);
        }
    }
}

File: Trip-Planner/src/app/classes/Gebruiker.ts
import { Plaats } from "./Plaats";
import { UserTrip } from "./UserTrip";

export class Gebruiker {
    userId: number;
    voornaam: string;
    achternaam: string;
    mail: string;
    wachtwoord: string;
    isAdmin: boolean;
    isActief: boolean;
    adres: String;
    plaats: Plaats;
    userTrips: UserTrip[] = [];


    constructor(userId: number, voornaam: string, achternaam: string, mail: string, wachtwoord: string, isAdmin: boolean, isActief: boolean, adres: String, plaats: Plaats) {
        this.userId = userId;
        this.voornaam = voornaam;
        this.achternaam = achternaam;
        this.mail = mail;
        this.wachtwoord = wachtwoord;
        this.isAdmin = isAdmin;
        this.isActief = isActief;
        this.adres = adres;
        this.plaats = plaats;
    }

}
